### PR TITLE
ref: Remove short version from release model

### PR DESCRIPTION
**Pull request by HazAT**
This is the server side part of https://github.com/getsentry/sentry/pull/13889

Talked to @mitsuhiko offline and we want to kill `short_version` of the release with 🔥 

Email:
image

### CODE REVIEW

File: src/sentry/api/serializers/models/commit.py [hunk 1]
```diff
@@ -87,7 +87,6 @@ def serialize(self, obj, attrs, user):
         data["releases"] = [
             {
                 "version": r.version,
-                "shortVersion": r.short_version,
```
Comment (mitsuhiko) [inline src/sentry/api/serializers/models/commit.py]: lets keep this as alias for `version` for external api consumers

**Review by mitsuhiko — CHANGES_REQUESTED** [review]

File: src/sentry/models/release.py [hunk 2]
```diff
@@ -228,19 +228,6 @@ def merge(cls, to_release, from_releases):
...
-    def short_version(self):
-        return Release.get_display_version(self.version)
-
-    @staticmethod
-    def get_display_version(version):
-        match = _dotted_path_prefix_re.match(version)
-        if match is not None:
-            version = version[match.end() :]
-        if _sha1_re.match(version):
-            return version[:7]
```
Comment (markstory) [inline src/sentry/models/release.py]: In the client side changes you're retaining the hash slicing. Would the client code be simpler if the API emitted the 'shortVersion' when the version name was hash shaped?

Comment (HazAT) [reply src/sentry/models/release.py]: I only kept the hash slicing for commit hashes and not for releases.
We want to get rid of the truncating release names entirely.

**Comment by HazAT** [comment]
Added @mnoble since the semantics of the API changes.

**Review by mitsuhiko — APPROVED** [review]

**Pull request merged**